compiler/rustc_span/src/hygiene.rs

```diff
@@ -938,7 +974,7 @@ pub struct ExpnData {
     pub local_inner_macros: bool,
     /// Should debuginfo for the macro be collapsed to the outermost expansion site (in other
     /// words, was the macro definition annotated with `#[collapse_debuginfo]`)?
-    pub(crate) collapse_debuginfo: bool,
+    pub collapse_debuginfo: bool,
```

Comment:
If the standard library crates are not hardcoded (https://github.com/rust-lang/rust/pull/118903#discussion_r1425574930), then `in_macro_expansion_with_collapse_debuginfo` can be moved back to this crate and the newly `pub`lic items in this file can become private again.

Reply:
`HygieneData` in particular is supposed to be private and it's better to organize things to not leak it to the outside in `walk_chain_collapsed` and related functions.